Objects/longobject.c

```diff
@@ -556,14 +556,28 @@ PyLong_AsLongAndOverflow(PyObject *vv, int *overflow)
         res = -1;
         i = _PyLong_DigitCount(v);
         sign = _PyLong_NonCompactSign(v);
-        x = 0;
+
+        digit *digits = v->long_value.ob_digit;
+        assert(i >= 2);
+        /* unroll 1 digit */
+        --i;
+        assert(ULONG_MAX >= ((1UL << PyLong_SHIFT) - 1));
+        x = digits[i];
+
+        #if ((ULONG_MAX >> PyLong_SHIFT)) >= ((1UL << PyLong_SHIFT) - 1)
```

Comment:
We can also unroll the second digit for the other methods. It gains a bit of performance at the cost of a bit of extra code. The unrolling can be factored out to a helper function (the loop itself is not so easy because there are small differences in how to handle the overflow)